esphome/components/packet_transport/packet_transport.h

```diff
@@ -5,7 +5,7 @@
 #ifdef USE_SENSOR
 #include "esphome/components/sensor/sensor.h"
 #endif
-#ifdef USE_BINARY_SENSOR
+#if defined(USE_BINARY_SENSOR) || defined(USE_STATUS_SENSOR)
```

Comment:
```suggestion
#ifdef USE_STATUS_SENSOR
```
USE_BINARY_SENSOR must be defined if there is a binary sensor configured.

Also add USE_STATUS_SENSOR to core/defines.h

Reply:
I'm not sure I agree.
Originally the line was `#ifdef USE_BINARY_SENSOR`
For that `esphome/components/binary_sensor/binary_sensor.h` was included to be used further down. That should still be the case.
For the status sensor the same base-definitions are used, and hence the include is needed.

Reply:
Sorry, should be the other way around - `#ifdef USE_BINARY_SENSOR`, i.e. what was there originally, is all that is required here (my comment was correct, just not the edit.) If USE_BINARY_SENSOR is not defined, you can't include the file.

Reply:
Ok, reverted it to as originally, and depend on that adding a binary status sensor will automatically add the define USE_BINARY_SENSOR